### ISSUE TITLE
2019Q3 - 3.2 Develop regular QA process for manual verification of deployed interventions and do it once

### ISSUE THREAD

**Issue by miketaylr** [mozilla/webcompat-team-okrs#91]
Ensure deployed website interventions in Firefox are still required and providing the expected result for the user.

**Comment by softvision-oana-arbuzov** [comment]
We've performed testing on the UA overrides and Interventions from `about:compat` on Desktop. Bellow are our results:

*Note: 
    **Not Fixed** = still broken while the UA override/Intervention is disabled
    **Fixed** = fixed regardless if UA override/Intervention is enabled or not
    **N/A** = not available/not tested

**[Environment]**
Browser / Version: Firefox Nightly 70.0a1 (2019-07-31)
Operating System: Windows 10 Pro, MacOS 10.14.5

**User Agent Overrides (about:compat Desktop):**
[Not Fixed] **rolb.santanderbank.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1563839
- https://github.com/webcompat/web-bugs/issues/35386 

**Interventions (about:compat Desktop):**
[Not Fixed] **Ib.absa.co.za**
- https://bugzilla.mozilla.org/show_bug.cgi?id=1452707 
- https://github.com/webcompat/web-bugs/issues/16401 

[Not Fixed] **histography.io** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1457335
- https://github.com/webcompat/web-bugs/issues/1804 
- https://github.com/webcompat/web-bugs/issues/20523 

[Not Fixed] **bankofamerica.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1472075
- https://bugzilla.mozilla.org/show_bug.cgi?id=1280834
- https://github.com/webcompat/web-bugs/issues/2787
- https://github.com/webcompat/web-bugs/issues/9661

[N/A - no elections available] election.gov.np 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1472081

[Not Fixed] **portalminasnet.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1482066
- https://github.com/webcompat/web-bugs/issues/18143 

[Not Fixed] **sreedharscce.in** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1526977
- https://github.com/webcompat/web-bugs/issues/21505 

[Not Fixed] **dns.google.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1567610
- https://github.com/webcompat/web-bugs/issues/22494

[Not Fixed] **twitch.tv** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1518781
- The issue is still reproducible with the Intervention disabled.

[Fixed] **gaming.youtube.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1305028
- The layout has changed. There is now only one scrollbar http://prntscr.com/omxxsv

[Not Fixed] **discordapp.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1432935
- The white vertical lines are still displayed, with the intervention disabled. http://prntscr.com/omyjvu

[N/A] **breitbart.com** 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1432935
- Not sure what the issue is about, BZ1432935 is a general bug. Looks like Chrome, with the Intervention disabled.

We will perform another test both on Desktop and Mobile, by the end of this OKR.

**Comment by denschub** [comment]
@softvision-oana-arbuzov We're pending some changes in the system addon that I'd like to have very careful testing on. We're currently blocked by another Firefox for Android-related issue ([bug 1570690](https://bugzilla.mozilla.org/show_bug.cgi?id=1570690)), but as soon as that is resolved, I want to land https://github.com/mozilla/webcompat-addon/pull/74 and with that, I want us to confirm all active interventions just as a safety net.

Could we coordinate here, so that you (or someone else) can do that testing as part of this goal? :) I don't have a timeline yet, since we're blocked on that other issue, but I'll keep you updated!

**Comment by softvision-sergiulogigan** [comment]
@denschub, Oana is on PTO starting tomorrow until the 18th, but me or @cipriansv would be happy to help!

**Comment by denschub** [comment]
Oh, it's not an immediate action point! I will reach out to all of you when we're ready to start, I just wanted to provide an early heads-up :)

**Comment by miketaylr** [comment]
Update: waiting until the end of the quarter to finish this up.

**Comment by miketaylr** [comment]
Update: should finish up next week.

**Comment by softvision-sergiulogigan** [comment]
The QA process for manual verification of deployed interventions - https://wiki.mozilla.org/Compatibility/UA_Override_&_Interventions_Testing

**Comment by cipriansv** [comment]
We have performed testing on the **UA overrides and Interventions** from **about:compat** on **Desktop** and **Mobile**. The results are the following:

**Testing Environment:**
Browser / Version: Firefox Nightly Mobile 68.2a1 (2019-09-23), Firefox Nightly 71.0a1(2019 -09-24)
Operating System: Huawei P10 (Android 9.0) - 1080 x 1920 pixels (~432 ppi pixel density), Windows 10 Pro, macOS 10.14.6

**Desktop Interventions and UA Overrides:**

- **Still reproducible** having the **User-Agent Overrides** disabled: 

1. [rolb.santanderbank.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1563839) – the input field labels are not displayed

- **Still reproducible** having the **Interventions** disabled:

1. [ib.absa.co.za ](https://bugzilla.mozilla.org/show_bug.cgi?id=1452707)– still receiving the “Browser unsupported” message
2. [histography.io](https://bugzilla.mozilla.org/show_bug.cgi?id=1457335) - still receiving the “Try Chrome or Safari” message
3. [bankofamerica.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1472075) - still receiving the “You are using an unsupported browser version” message
4. [www.election.gov.np](https://bugzilla.mozilla.org/show_bug.cgi?id=1472081) - "Page Not Found" error message displayed
5. [www.sreedharscce.in](https://bugzilla.mozilla.org/show_bug.cgi?id=1526977) -  the login form is still hiding on the top of the page
6. [discordapp.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1432935) – the white vertical line is still displayed
7. [twitch.tv](https://bugzilla.mozilla.org/show_bug.cgi?id=1518781)  -  the three-dotted menu button is not easily clickable
8. [dns.google.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1567610) – the page content is still shifted to the left side

- **Not reproducible** having the **Interventions** disabled:

1. [portalminasnet.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1482066) – you are now able to type the username and password in the corresponding fields
2. [gaming.youtube.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1305028) – the scrollbar is not overlapped by the video after reducing the browser window
3. [breitbart.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1432935) -  the horizontal scrollbar is not displayed anymore

- **Not reproducible** having the **User-Agent Overrides** disabled:
   none

**Mobile Interventions and UA Overrides:**

- **Still reproducible** having the **User-Agent Overrides** disabled: 

1. [m.imgur.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1480710) – unable to test due to shut down of the logged-in Imgur mobile web experience
2. [tieba.baidu.com](https://bugzilla.mozilla.org/show_bug.cgi?id=945963) – low-fi mobile content received 
3. [lohaco.jp](https://bugzilla.mozilla.org/show_bug.cgi?id=1177298) – the desktop version of the site received
4. [nhk.or.jp](https://bugzilla.mozilla.org/show_bug.cgi?id=1177298) -unable to test since Firefox Preview does not yet support UA overrides.
6. [directv.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1338260) – you are now able to scroll but the images are still cropped when the device is in landscape mode 
7. [rakuten.co.jp](https://bugzilla.mozilla.org/show_bug.cgi?id=1385206) – the desktop version of the site received 
8. [mobile.de](https://bugzilla.mozilla.org/show_bug.cgi?id=969844) – the desktop version of the site received
8. [cc.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1509831) – unsupported browser message still displayed when trying to play a video
9. [cineflix.com.br](https://bugzilla.mozilla.org/show_bug.cgi?id=1508516) – page content is not displayed 
10. [zmags.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1509873) – unsupported browser error message displayed 
11. [posts.google.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1566253) – 500 error message displayed 
12. [beeg.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1567945) – a blank page is received 

- **Still reproducible** having the **Interventions** disabled:

1. [mail.google.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1561371) – the mail content is not horizontally scrollable 
2. [dns.google.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1567610) – page content is shifted to the left
3. [zertificate.commerzbank.de](https://bugzilla.mozilla.org/show_bug.cgi?id=1568256) – the site is not correctly rendered 

- **Not reproducible** having the **User-Agent Override**s disabled 

1. [rolb.santanderbank.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1563839) – the input field labels are displayed 
2. [redbull.com](https://bugzilla.mozilla.org/show_bug.cgi?id=1509852) – the site is displayed 

- **Not reproducible** having the **Interventions** disabled:

1. [Sites using PDK 5 video](https://bugzilla.mozilla.org/show_bug.cgi?id=1551672) – the video plays having the intervention disabled

**Comment by miketaylr** [comment]
Thanks @softvision-oana-arbuzov @softvision-sergiulogigan @cipriansv for all your work here. I'd like us to do this going forward, about once a month.

@ksy36 can we get a bug on file to remove the interventions that are no longer needed? Thanks!

**Comment by ksy36** [comment]
Created https://bugzilla.mozilla.org/show_bug.cgi?id=1585609
